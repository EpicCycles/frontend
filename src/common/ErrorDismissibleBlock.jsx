import React, {Component} from 'react'
import {Message} from 'semantic-ui-react'

class ErrorDismissibleBlock extends Component {
    render() {
        const { application, removeMessage } = this.props;
        const messageColor = (application.messageType && (application.messageType === 'E')) ? 'red' : 'green';

        return <Message color={messageColor}
                        content={application.message}
                        onDismiss={removeMessage}
        />
    };
}

export default ErrorDismissibleBlock;
