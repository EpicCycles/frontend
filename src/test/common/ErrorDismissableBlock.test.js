import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorDismissibleBlock from '../../common/ErrorDismissibleBlock';

describe('ErrorDismissibleBlock tests', () => {
  it('renders the errorDismissableBlock correctly', () => {
    const application = { message: 'show me the error', messageType: 'E' };
    const errorDismissibleBlock = shallow(
      <ErrorDismissibleBlock application={application} removeMessage={() => {}} />,
    );
    expect(toJson(errorDismissibleBlock)).toMatchSnapshot();
  });
});
