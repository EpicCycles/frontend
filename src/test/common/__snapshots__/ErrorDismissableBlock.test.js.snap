// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDismissibleBlock tests renders the errorDismissableBlock correctly 1`] = `
<Message
  color="red"
  content="show me the error"
  onDismiss={[Function]}
/>
`;
